### PR TITLE
[DOC] Repair format of What's Here sections in object.c

### PR DESCRIPTION
**Pull request by BurdetteLamar**

### CODE REVIEW

File: object.c
Comment (nobu) [inline object.c:4199]:
```suggestion
 *  - #!: Returns the boolean negation of +self+: +true+ or +false+.
 *  - #!=: Returns whether +self+ and the given object are _not_ equal.
```

File: object.c
Comment (nobu) [inline object.c:4246]:
```suggestion
 *  - #!~: Returns +true+ if +self+ does not match the given object,
 *    otherwise +false+.
 *  - #<=>: Returns 0 if +self+ and the given object +object+ are the same
 *    object, or if <tt>self == object</tt>; otherwise returns +nil+.
```

**Review by nobu — CHANGES_REQUESTED** [review]

File: object.c
Comment (nobu) [inline object.c:4461]: I think the canonical notation is `` #` `` here, and `` command` `` is extraneous.
Do you think `` #`command` `` should be parsed as single method reference?

File: object.c
Comment (BurdetteLamar) [reply object.c:4199]: The suggested revisions above do not become links on my system:
- $ ruby --version
  ruby 3.1.0p0 (2021-12-25 revision fb4df44d16) [x64-mingw-ucrt]
- $ rdoc --version
  6.4.0

Is this fixed in later versions?

There are many more like this in the What's Here sections.  When they are no longer needed, I'll be happy to do a sweep through to clean them up.

File: object.c
Comment (BurdetteLamar) [reply object.c:4461]: I'm agnostic on this one.  In the What's Here section, <tt>#`</tt> in source would become a link with just the one character <tt>`</tt>, a tiny target even if correct.  How would you like this to be?

Also, <tt>#`</tt> did not link on my machine (versions same as above).

**Comment by BurdetteLamar** [comment]
@nobu, can we leave these links as-is for now, and deal with all such (in all classes) all at once, and later?  If so, I'll still need an approval from you or @jeremyevans.

File: object.c
Comment (nobu) [reply object.c:4199]: It is fixed at https://github.com/ruby/rdoc/pull/865 and has been merged to the ruby master, but rdoc is not released yet.

**Comment by nobu** [comment]
Do you mean to change just the method list styles?
I'm ok for it.

**Comment by BurdetteLamar** [comment]
> Do you mean to change just the method list styles? I'm ok for it.

I mean to merge now, then fix after the fix is released.  Ok?

**Comment by BurdetteLamar** [comment]
When the fix at 385 is released, I will sweep through all classes that should make use of it.

**Comment by BurdetteLamar** [comment]
Sorry, @nobu, I did not understand you about merging this;  now I do.

**Pull request merged**
